### PR TITLE
unit-tests: explicitly create the home directory

### PR DESCRIPTION
**Pull request by allisonkarlitskaya**
`adduser` recently changed its behaviour for system users, not to create
the home directory.  We expect the home directory to be created, though
(since we build in it), so explicitly specify `--create-home`.

### CODE REVIEW

**Comment by jelly** [comment]
FYI: comes from https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=679746

**Review by jelly — DISMISSED** [review]

**Comment by allisonkarlitskaya** [comment]
The failing unit tests are because of clang, not i386 (I tested gcc/i386 and clang/amd64 locally).

That points to https://bugs.kde.org/show_bug.cgi?id=452758  (thanks @jelly).

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -6,7 +6,9 @@ jobs:
     strategy:
       matrix:
         startarg:
-          - { make: 'check-memory', cc: 'clang', tag: 'i386' }
+          # new Clang breaks Valgrind, so use gcc for all `check-memory`
+          # scenarios for now: https://bugs.kde.org/show_bug.cgi?id=452758
```
Comment (jelly) [inline .github/workflows/unit-tests.yml:10]: Alternatively:
```
            # HACK: valgrind does not work with DWARF5 symbols generated by Clang 14 https://bugs.kde.org/show_bug.cgi?id=452758
            command = ['sh', '-c', 'CFLAGS="-O2 -gdwarf-4" /source/autogen.sh; exec make -j$(nproc) ' + shlex.quote(args.make)]
```

Comment (allisonkarlitskaya) [reply .github/workflows/unit-tests.yml:10]: ...thus also avoiding the test name changing.  Sounds good :)

**Comment by martinpitt** [comment]
Hmmm.. the tests succeeded here, but they still fail in the refresh run: https://github.com/cockpit-project/cockpit/runs/8189281079?check_suite_focus=true

**Pull request merged**